src/Schema/ColumnEditor.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Schema;
+
+use Doctrine\DBAL\Platforms\SQLServerPlatform;
+use Doctrine\DBAL\Schema\Exception\InvalidColumnDefinition;
+use Doctrine\DBAL\Schema\Name\UnqualifiedName;
+use Doctrine\DBAL\Types\Exception\TypesException;
+use Doctrine\DBAL\Types\Type;
+
+final class ColumnEditor
+{
+    private ?UnqualifiedName $name = null;
+
+    private ?Type $type = null;
+
+    private ?int $length = null;
+
+    private ?int $precision = null;
+
+    private int $scale = 0;
+
+    private bool $unsigned = false;
+
+    private bool $fixed = false;
+
+    private bool $notNull = true;
+
+    private mixed $defaultValue = null;
+
+    private mixed $minimumValue = null;
+
+    private mixed $maximumValue = null;
+
+    private bool $autoincrement = false;
+
+    private string $comment = '';
```

Comment:
Not a blocker/ might need to be done separately:

Now that we have mostly "typed enough" properties (as in, they don't need extra phpdoc to convey the needed type information), maybe we should start removing empty lines between each property?

Reply:
I don't have a strong preference for whether to keep spaces here or not. I'm fine if it becomes part of the coding standard, and we format our code accordingly. Otherwise, it may be challenging to keep this code consistently formatted.